pbs_nodefile.c

```diff
@@ -0,0 +1,144 @@
+/*
+ * Plugin to set the PBS_NODEFILE env variable and its contents for the job
+ *
+ * (C) 2018 Ghent University
+ *
+ * Author: Andy Georges
+ *
+ * Loosely based on Spank plugin private-tmpdir (c) HPC2N.umu.se
+ *
+ */
+
+/* Needs to be defined before first invocation of features.h so enable
+ * it early. */
+#define _GNU_SOURCE        /* See feature_test_macros(7) */
+#define _XOPEN_SOURCE 500
+
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+#include <slurm/spank.h>
+#include <unistd.h>
+
+SPANK_PLUGIN(env-test, 1);
+
+// Gobals
+static char *generate_pbsnodefile_command;
+int init_opts = 0;
+int init_complete = 0;
+
+
+int slurm_spank_init(spank_t sp, int ac, char **av)
```

Comment:
there is no cleanup of the file when the job completed/failed/....?

Reply:
Of course there is. It's in the local tmp. Everything there is destroyed.

Reply:
? only if you use the private tmpdir plugin. if you can, make this plugin standalone and cleanup the file